hazelcast/src/main/java/com/hazelcast/internal/usercodedeployment/UserCodeDeploymentService.java

```diff
@@ -69,7 +69,13 @@ public void init(NodeEngine nodeEngine, Properties properties) {
 
     public void defineClasses(List<Map.Entry<String, byte[]>> classDefinitions) {
         if (!enabled) {
-            throw new IllegalStateException("User Code Deployment is not enabled.");
+            throw new IllegalStateException("User Code Deployment is not enabled. "
+                    + "To enable user code deployment, please do one of the following:\n"
+                    + "- Change member config using JAVA API: "
+                    + "config.getUserCodeDeploymentConfig().setEnabled(true);\n"
+                    + "- Change XML/YAML configuration property: hazelcast.user-code-deployment.enabled to true\n"
+                    + "- Add system property: -Dhz.user-code-deployment.enabled=true\n"
+                    + "- Add environment variable property: HZ_USERCODEDEPLOYMENT_ENABLED=true");
```

Comment:
it is not clear to me if this is a java environment variable or something for containers?

Reply:
I actually used the same structure of the error message that it had for REST API.

But what do you mean? Environment variable is an environment variable, no? There's just one configured directly in the OS, there's nothing like "Java environment variable", right?

Reply:
yes java has it: https://docs.oracle.com/javase/tutorial/essential/environment/env.html#:~:text=Like%20properties%20in%20the%20Java,also%20between%20command%20line%20interpreters.

Reply:
Maybe `property` next to `environment variable` is confusing because `property` usually relates to Java system properties. What do you think about removing it? 

```suggestion
                    + "- Add environment variable: HZ_USERCODEDEPLOYMENT_ENABLED=true");
```

Reply:
i think we can also say where we commonly expect to see `HZ_USERCODEDEPLOYMENT_ENABLED`.  

`- Add environment variable(for containerized apps): HZ_USERCODEDEPLOYMENT_ENABLED=true")`

Reply:
This is not exclusive to containerized apps. You can configure the member through environmental variable even if you're running on a bare metal.

+1 to removing the "property" word though.

Reply:
>You can configure the member through environmental variable even if you're running on a bare metal.

In this case, should i expect to find a place in the codebase where we read `HZ_USERCODEDEPLOYMENT_ENABLED` environment variable? At the moment i couldn't find it.

Reply:
No, you can't find it. This is done automatically for all the configuration through configuration override: https://docs.hazelcast.com/hazelcast/5.0/configuration/overriding-configuration-settings

Reply:
I wasn't aware we have his auto conversion, i see now. Then i think it is sufficient removing the `property` word. Thanks for the explanation.